src/components/Search/Search.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { EventHandler, FormEventHandler, useState } from "react";
+type IntrinsicElements = JSX.IntrinsicElements["input"];
+
+interface Props extends IntrinsicElements {
+  variation?: SearchVariation;
+  onSearch?: EventHandler<any>;
+  onInput?: FormEventHandler<HTMLInputElement>;
+  disabled?: boolean;
+  initialValue?: string;
+  placeholder?: string;
+  labelText?: string;
+}
+type SearchVariation = "default" | "big" | "small";
+
+/**
+ * Search Component
+ * @param {string}            variation             Renders the style of the search component.
+ * @param {EventHandler}      onSearch              Handles its behavior when the search button is clicked or the user presses enter in the input. Returns value of the input. Returns value of the input.
+ * @param {EventHandler}      onInput               Handles its behavior when the user is typing.
+ * @param {boolean}           [disabled]            Determines whether or not a button is enabled.
+ * @param {string}            [initialValue]        If provided, used as value of search input.
+ * @param {string}            [placeholder]         The text placeholder for this component.
+ * @param {string}            [labelText]           Label text for screen reader.
+ */
+
+export const Search: React.FC<Props> = ({
+  variation = "default",
+  disabled = false,
+  onSearch,
+  initialValue = "",
+  labelText = "search",
+  onInput,
+  ...rest
+}) => {
+  const [inputValue, setInputValue] = useState<string>(initialValue);
+
+  return (
+    <section aria-label="Search component">
+      <form
+        className={`usa-search role usa-search--${variation}`}
+        role="search"
+      >
+        <label className="usa-sr-only" htmlFor="search-field">
+          {labelText}
+        </label>
+        <input
+          className="usa-input"
+          id="search-field"
+          type="search"
+          name="search"
+          value={inputValue}
+          disabled={disabled}
+          onInput={onInput}
+          onChange={(e) => setInputValue(e.target.value)}
+          onKeyDown={(e) => {
+            if (e.key === "Enter") {
+              e.preventDefault();
+              onSearch && onSearch(inputValue);
+            }
+          }}
+          {...rest}
+        />
+        <button
+          className="usa-button padding-y-0 padding-x-2"
+          type="button"
+          onClick={() => onSearch && onSearch(inputValue)}
```

Comment:
```suggestion
          onClick={() => onSearch(inputValue)}
```
This is the only line uncovered in test coverage. I'm assuming that it's just upset that there's not a test case where the function is not present. Given that this is a search box I'm in favor of making `onSearch()` required. It simplifies this slightly and brings the coverage up.